src/DocBlock/TypeExpression.php

```diff
@@ -53,7 +53,7 @@ final class TypeExpression
                     (?<array_shape_start>(?i)(?:array|list|object)(?-i)\h*\{\h*)
                     (?<array_shape_inners>
                         (?<array_shape_inner>
-                            (?<array_shape_inner_key>(?:(?&constant)|(?&identifier))\h*\??\h*:\h*|)
+                            (?<array_shape_inner_key>(?:(?&constant)|(?&identifier)|(?&name))\h*\??\h*:\h*|)
```

Comment:
the backtrack limit should be probably fixed by `*+` on l62

also `|(?&name)` seems to be wrong in sense of phpdoc grammar/theory - how is this exactly parsed by https://github.com/phpstan/phpdoc-parser?